libr/util/diff.c

```diff
@@ -54,8 +54,8 @@ R_API int r_diff_buffers_static(RDiff *d, const ut8 *a, int la, const ut8 *b, in
 		} else {
 			if (hit>0) {
 				struct r_diff_op_t o = {
-					.a_off = d->off_a+i-hit, .a_buf = a+i-hit, .a_len = la,
-					.b_off = d->off_b+i-hit, .b_buf = b+i-hit, .b_len = lb 
+					.a_off = d->off_a+i-hit, .a_buf = a+i-hit, .a_len = hit,
```

Comment:
let's assume you have no difference at byte 1 but at byte 2. Both files have a size of 2. So if you now run this stuff, and get to this point i is 2, hit is 1, la and lb are 2. a_off and b_off both get set to 1 (i-hit), and a_len and b_len both get set to 2 in the old version. both buffers point to the second byte of a/b (&a[1] or &b[1]). So now you have 2 buffers, of the size of 2 and a pointer for each of them, that points to their second byte, and now you start reading at these pointers 2 bytes. Question: Is this a valid read?